torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -245,9 +245,70 @@ void ComputeAtMap::build(Fusion* fusion, GpuLower* gpu_lower) {
     }
 
     auto tv_outputs = ir_utils::filterByType<TensorView>(expr->outputs());
+    TensorView* first_output_tv = nullptr;
     for (auto c_tv : tv_outputs) {
       consumer_tvs.push_back(c_tv);
 
+      if (first_output_tv == nullptr) {
+        first_output_tv = c_tv;
+      } else {
+        // Map multi outputs of an expression to eachother. c is current output,
+        // and f as first output. Keep consistent with the later section of
+        // producer and consumers. Which here producer is now "first output",
+        // and consumer is still consumer.
+
+        TORCH_INTERNAL_ASSERT(
+            c_tv->getRootDomain().size() ==
+                first_output_tv->getRootDomain().size(),
+            "Multiple outputs with mismatched dimensions is not supported. ",
+            "Only supported case is welford op where all outputs tvs have idential domains.");
+        // p->f, c->c
+        std::unordered_map<IterDomain*, IterDomain*> c2f_root_map;
+        for (size_t i = 0; i < first_output_tv->getRootDomain().size(); i++) {
+          c2f_root_map.insert(std::make_pair(
+              c_tv->getRootDomain()[i], first_output_tv->getRootDomain()[i]));
+        }
+
+        // Multi output mapping
+        auto replay_FasC = BestEffortReplay(
+            first_output_tv->domain()->domain(),
+            c_tv->domain()->domain(),
+            c2f_root_map);
+
+        auto c2f_map = replay_FasC.getReplay();
+
+        // If we're creating parallel map, only map the leaf
+        // axes. Also, the producer axis must be left of the CA
+        // point.
+        // Otherwise, map the entire replay map.
+        if (mapping_mode_ == MappingMode::PARALLEL) {
+          // Mark axes left of compute at point for parallel type tracking
+          std::unordered_set<IterDomain*> producer_axes_to_map(
+              first_output_tv->domain()->domain().begin(),
+              first_output_tv->domain()->domain().begin() +
+                  first_output_tv->getComputeAtPosition());
+
+          for (auto c_id : c_tv->domain()->domain()) {
```

Comment:
nit: probably not necessary, but how about checking if `c_id` is also left of the CA position? I think it's enforced since we check the position of the first consumer. Am I right?

Reply:
Yes, you're right. We're also enforcing now that all outputs of an expression are replayed in the same exact way during the compute at pass. Doesn't matter in this replay if it's to the CA position on each output or the entire domain as it would result in the same thing as that's how the "first" output gets its transformations anyways.